### ISSUE TITLE
react-native Support

### ISSUE THREAD

**Issue by AlanFoster** [softonic/axios-retry#20]
Hey; I've just been checking out using axios-retry with a react-native client.

Axios 2.x worked great, but no longer works with the new 3.x release.

I pinned it down to this predicate now failing:

```js
    && Boolean(error.code) // Prevents retrying cancelled requests
```

For visibility, the react-native axios object within an error scenario was:

```
// [Error: Network Error]
{ 
  line: 54218,
  column: 24,
  sourceURL: 'http://10.42.54.44.xip.io:8081/index.ios.bundle?platform=ios&dev=true&minify=false',
  config: {
     adapter: [Function: xhrAdapter],
     transformRequest: { '0': [Function: transformRequest] },
     transformResponse: { '0': [Function: transformResponse] },
     timeout: 0,
     xsrfCookieName: 'XSRF-TOKEN',
     xsrfHeaderName: 'X-XSRF-TOKEN',
     maxContentLength: -1,
     validateStatus: [Function: validateStatus],
     headers: {
      Accept: 'application/json'
     },
     baseURL: 'https://example.com',
     method: 'get',
     params: {
      foo: 'bar'
     },
     url: 'https://example.com/foo',
     data: undefined,
     'axios-retry': {
      retryCount: 0
     }
  },
  request: {
    UNSENT: 0,
     OPENED: 1,
     HEADERS_RECEIVED: 2,
     LOADING: 3,
     DONE: 4,
     readyState: 4,
     status: 0,
     timeout: 0,
     withCredentials: true,
     upload: {},
     _aborted: false,
     _hasError: true,
     _method: 'GET',
     _response: 'The network connection was lost.',
     _url: 'https://example.com/foo?foo=bar',
     _timedOut: false,
     _trackingName: 'unknown',
     _incrementalEvents: false,
     responseHeaders: undefined,
     _requestId: null,
     _cachedResponse: undefined,
     _headers: {
        accept: [Getter/Setter]
      },
     _responseType: '',
     _sent: true,
     _lowerCaseResponseHeaders: {},
     _subscriptions: []
   },
  response: undefined
}
```

Thankfully the API for axios-retry is pretty good as it allows you to provide an override for this check, but it would be interesting to see if there's a way to update axios-retry to provide support both react-native and the browser :)

Any thoughts on this? 🤔

**Comment by rubennorte** [comment]
We added the condition to prevent unknown network errors being retried blindly but we forgot to remove the restriction for safe and idempotent requests (which is now the default). Fixing in a following PR.

**Comment by rubennorte** [comment]
Fixed and released in 3.0.1
